### PR TITLE
Provide improved flavors' support

### LINKED ISSUE #77: Support mulitple flavors in recording and comparing screenshots
Currently only one flavor is possible, which is set in build.gradle.
This is an enhancement idea to support multiple flavors:
- pass option via CLI, 
  - .eg. :-Pshot.flavor=name
  - overrides setting in build.gradle
- store screenshots in /screenshots/flavor/ to support

### PR DESCRIPTION
**Pull request by pedrovgs**
### :pushpin: References
* **Issue:** Closes #77

### :tophat: What is the goal?

Provide idiomatic flavors support for android projects by creating a generic task associated to different tasks configured by flavors and build type. Based on this, we've simplified a lot the plugin configuration and now we can execute our tests for any flavor and build type configuration easily.

### How is it being implemented?

To implement this feature we've followed these steps:

* Update the Facebook dependency being used to 0.12.0.
* Add a the android gradle plugin as a dependency so we can create every plugin task based on the project variant.
* Modify the project's folder configuration to be based on the build type and the flavor. Now the screenshots are saved in different directories based on the build configuration. I don't like how we pass the build type and flavor information around and how the folders' path is created, but I don't have time to improve it for now.
* Update the CI configuration to run the tests in the consumer created as an example of how to use flavors.
* Simplify the composer support.
* Prepare the code to release a new version.
* Update the project documentation.

### How can it be tested?

🤖

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -34,25 +34,26 @@ Setup the Gradle plugin:
...
-    appId = 'YOUR_APPLICATION_ID'
-  }
 ```
 
 This plugin sets up a few convenience commands you can list executing ``./gradlew tasks`` and reviewing the ``Shot`` associated tasks:
 
-**If you are using flavors update your shot configuration inside the ``build.gradle`` file as follows:**
+![shotTasksHelp](./art/tasksDescription.png)
+
+**If you are using flavors the available Shot gradle tasks will be configured based on your flavors and build types configuration.** You can find all the available shot tasks by executing ``./gradlew tasks``. For example, if you app has two flavors: ``green`` and ``blue`` the list of available Shot tasks will be:
```
Comment (Serchinastico) [inline README.md:47]:
```suggestion
**If you are using flavors the available Shot gradle tasks will be configured based on your flavors and build types configuration.** You can find all the available shot tasks by executing ``./gradlew tasks``. For example, if your app has two flavors: ``green`` and ``blue`` the list of available Shot tasks will be:
```

File: shot-consumer-flavors/scripts/create_emulator.sh [hunk 2]
```diff
@@ -1,24 +1,24 @@
 #!/bin/bash
 
-if $ANDROID_HOME/tools/android list avd | grep -q Nexus_5X_API_26; then
```
Comment (Serchinastico) [inline shot-consumer-flavors/scripts/create_emulator.sh:3]: What's the reason behind the changes in the emulator and the SDCard image names?

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala [hunk 3]
```diff
@@ -61,28 +58,76 @@ class ShotPlugin extends Plugin[Project] {
...
+    val appExtension =
+      project.getExtensions.getByType[AppExtension](classOf[AppExtension])
+    val baseTask = project.getTasks.create(
+      Config.defaultTaskName,
+      classOf[ExecuteScreenshotTestsForEveryFlavor])
+    appExtension.getApplicationVariants.all { variant =>
+      val flavor = variant.getMergedFlavor
+      val completeAppId = flavor.getApplicationId + Option(
+        flavor.getApplicationIdSuffix).getOrElse("")
+      if (variant.getBuildType.getName != "release") {
```
Comment (Serchinastico) [inline shot/src/main/scala/com/karumi/shot/ShotPlugin.scala:70]: Interesting take, why do we prevent `release` builds to have a Shot task? Was this causing any issue or is it just to prevent people from running Shot in release builds? I think they can bypass this check by just renaming the release config on their side if that's the case.

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala [hunk 4]
```diff
@@ -61,28 +58,76 @@ class ShotPlugin extends Plugin[Project] {
...
 
   private def addTasks(project: Project): Unit = {
+    val appExtension =
+      project.getExtensions.getByType[AppExtension](classOf[AppExtension])
+    val baseTask = project.getTasks.create(
+      Config.defaultTaskName,
+      classOf[ExecuteScreenshotTestsForEveryFlavor])
+    appExtension.getApplicationVariants.all { variant =>
+      val flavor = variant.getMergedFlavor
+      val completeAppId = flavor.getApplicationId + Option(
```
Comment (Serchinastico) [inline shot/src/main/scala/com/karumi/shot/ShotPlugin.scala:68]: It's great we don't have to pass the app id anymore 👏

File: shot-consumer-flavors/scripts/create_emulator.sh (hunk 2 above)
Comment (pedrovgs) [reply shot-consumer-flavors/scripts/create_emulator.sh:3]: Just to be able to create different emulators with different sdcards for different consumers.

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala (hunk 3 above)
Comment (pedrovgs) [reply shot/src/main/scala/com/karumi/shot/ShotPlugin.scala:70]: Just because there are no tasks to test any release build. If you check the ``androidTests`` tasks for your project you'll notice the release variant has no task associated.

File: shot/src/main/scala/com/karumi/shot/ShotPlugin.scala (hunk 4 above)
Comment (pedrovgs) [reply shot/src/main/scala/com/karumi/shot/ShotPlugin.scala:68]: Yeap, I found this method looking at the android gradle plugin code and I decided to remove the config param 😃
